Adding support for bootmode 'default'

m4ce opened the pull request:
When bootmode is default, only the boot entry set as default will be considered when applying changes.

There was also an old post at https://www.redhat.com/archives/augeas-devel/2013-May/msg00006.html asking how to achieve this.

raphink commented:
Thank you.

This change breaks the unit tests and is lacking additional tests.

Please add missing tests and fix existing ones.

m4ce commented:
Hi,

I've adjusted the unit tests and added the missing ones. This also led me to change the logic of how the kernel parameters are categorized as 'all', 'recovery', 'default' and 'normal'.

In the legacy grub, the current logic is:
```
if the parameter happens to be in the default entry AND all the recovery entryies AND all the normal entries, then 'all'
else if the parameter happens to be in at least one recovery entry, then 'recovery'
else if the parameter happens to be in the default entry, then 'default'
else 'normal'
```

In the grub2 provider, 'default' is just another alias for 'normal'.

raphink commented:
Sorry for taking so long to review this @m4ce !

Could you update `README.md` with the new parameter and explanation on the fixed logic please?

raphink commented:
@m4ce any news?

m4ce commented:
@raphink,

sorry for not having come back to you in so long. I've updated the README.md to reflect the changes.

Happy holidays!

Cheers,
Matteo

raphink commented:
Rebased and merged as 2c47e358acbcdd99c35554a5e089aa10b709c6ba
